go.go

```diff
@@ -687,6 +781,54 @@ const goUnmarshalField = `{{if eq .Type "bool"}}
 		i++
 	}
 {{else if eq .Type "int32"}}
+{{- if .TypeList}}
+	if header == {{.Index}} {
+	{{template "unmarshal-varint" .}}
+		if x > uint(ColferListMax) {
+			return 0, ColferMax(fmt.Sprintf("colfer: {{.String}} length %d exceeds %d elements", x, ColferListMax))
+		}
+
+		l := int(x)
+
+		if end := i + l*4; end >= len(data) {
+			i = end
+			goto eof
+		}
+		a := make([]int32, l)
+		for ai := range a {
+			if i+1 >= len(data) {
+				i++
+				goto eof
+			}
+
+			x := uint32(data[i])
```

Comment:
Could use `uint` instead of `uint32` here and on line 810.

Reply:
changed